### DISCUSSION TITLE
No warning when closing out or navigating from current page when there is text in draft form

### DISCUSSION THREAD

**Discussion by JacksonCampbell99** [community/community#5764] [Other Feature Feedback, Questions, & Ideas]
Something about GitHub that has always surprised me is how they don't have any warning message come up when navigating or closing out of pages with drafts of comments or posts. For any site with unsaved changes and many social media pages, drafted posts and unsaved changes trigger a browser message saying "Leave site? Changes you made may not be saved." On most sites or apps including some social media, closing a post box that you have drafted text in brings up a message similar to "Discard this post?" with options like `Yes`, `No`, and `Save`. GitHub doesn't have that in the app or browser. There aren't many drafts in life I get too worried about losing, but accidentally closing a GitHub page with half a report written is one of them. Windows is good (terrible) at misinterpreting touch input to select a tab for clicking `X` on a tab and closing it. I have to be especially careful knowing that I could easily lose a lot of work. Could you please enable the site and app to sense a click to navigate within the website (navigate back in the app) or close a tab (swipe a comment down too far in the app) and pop up a warning of unfinished posts or comments to protect losing work? I'd also like to request a "save draft" feature integrated with the warning (a draft feature requires a pop-up before `X`ing out of something anyway); the draft on server aspect would make it simple to switch from app to site to upload a file/do something not available on the app or to set down and pick up a post or report later when convenient or from the convenience of a different device. To get an example, you can test the "save draft" warning dialogue on Twitter by closing out a tweet draft and you can test the warning message suggestion on Facebook by closing out a post for their in-site or in-app warning and closing a browser tab with a post draft for the browser warning (see screenshots for both). An example of an auto-draft/auto-save feature would be in-browser email drafting and creating feedback in the Windows Feedback Hub, which auto-saves if navigate outside the entry window or close out the application.

Case in point: my computer crashed while I was editing this discussion, and I had to rewrite a bunch of it. Thankfully, Edge happened to keep some of the cache on tab reload which is hit or miss.

![Twitter web: in-progress tweet warning/save tweet dialogue](https://user-images.githubusercontent.com/76508020/134355587-868d0ffa-b199-437f-b135-20ed007e3c9e.png)
![Twitter mobile: in-progress tweet/save tweet as draft dialogue](https://user-images.githubusercontent.com/76508020/134356470-1e87250a-6a35-4583-acf0-110f67f17a79.png)
![Facebook web: browser/in-site integrated warnings](https://user-images.githubusercontent.com/76508020/134355374-0ae3c677-f8d7-4722-85c1-524c4d08a445.png)
![Facebook mobile: in-progress warning/save post as draft dialogue](https://user-images.githubusercontent.com/76508020/134356879-479471d8-6f52-4bc7-8afb-cfc2a2fdf9ab.png)

**Comment by JacksonCampbell99** [comment]
Anybody going to respond?

**Reply by JacksonCampbell99** [reply]
Can I get a response please?

**Comment by gaknoll** [comment]
@JacksonCampbell99 sorry for the delay in getting a response! I appreciate you taking the time to thoroughly describe your feedback and potential solutions with examples. I agree with you that there is an opportunity here to improve the user experience around unpublished edits warning patterns across github.com. I have created a backlog issue for our Primer design system team who works on creating consistent patterns and components for our feature teams to use.

**Reply by JacksonCampbell99** [reply]
Thanks so much!
